### PR TITLE
Tcp transmission

### LINKED ISSUE #40: Integrate TCP Transmission into Plugin
Given the TCP client and server setup code is available
When the plugin collects data
Then the data should be packaged and transmitted over TCP using the Event Receiving Interface format

Developer Notes:
Integrate the TCP transmission code into the plugin so that the extracted data (altitude, airspeed, vertical velocity, and heading) is transmitted in real-time to the iMotions platform. Use the generate_packet() function to convert the collected data into the appropriate format before transmission.

Acceptance Criteria:

The plugin successfully transmits data over TCP when the flight simulator is running.
Data is formatted according to the Event Receiving Interface requirements.

### PR DESCRIPTION
**Pull request by carlyrhoover**
### Fixes #40 

---

### **What was changed?**

-plugin file 
---

### **Why was it changed?**

-added functionality to plugin start function that starts transmission and packet generation 

### **How was it changed?**

-Used new queue function and cpp multithreading to create a TCP instance within the plugin start function 
-Information is gathered from said queue and turned into packet
-resulting packet is then sent along the TCP transmission 

---

### **Screenshots (if applicable):**

- Add screenshots or logs that demonstrate the successful execution of the changes, such as flight data being logged or sent via TCP.

### CODE REVIEW

File: src/pilotdatasync-xp11.cpp [hunk 1]
```diff
@@ -102,6 +108,64 @@ void dummy_key_handler(
     int losing_focus
 ) {}
 
+int runTCPThread(){
```
Comment (aelnourSLU) [inline src/pilotdatasync-xp11.cpp:111]: The function has various integer returns, but do we have any particular meaning applied to each returned number?

File: src/pilotdatasync-xp11.cpp [hunk 2]
```diff
@@ -102,6 +108,64 @@ void dummy_key_handler(
...
+        return 1; 
+    }
+    
+    if (!client.createSocket()) {
+        std::cout << "Failed to create socket" << std::endl;
+        return 1; 
+        }
+        
+
+    std::string serverIP = "127.0.0.1";  // Changed to IPv4 localhost
```
Comment (aelnourSLU) [inline src/pilotdatasync-xp11.cpp:150]: We should consider updating this to the target address for iMotions soon

File: src/pilotdatasync-xp11.cpp [hunk 3]
```diff
@@ -102,6 +108,64 @@ void dummy_key_handler(
...
+    if (!client.connectToServer(serverIP)) {
+        std::cout << "Failed to connect to server" << std::endl;
+        return 1;
+    }
+    
+    // Send test messages
```
Comment (aelnourSLU) [inline src/pilotdatasync-xp11.cpp:156]: Can we get this switched to sending test packets using the `generate_packet` function I wrote in `threadig-tools.cpp`?

**Review by aelnourSLU — APPROVED** [review]
Overall looks good, the suggestions I made can be made either this sprint or next as needed

**Comment by harshitha-thota** [comment]
while running the code I am getting below error.

C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\TCPClient.cpp(41,30): warning C4244: '=': conversion from 'SOCK
ET' to 'int', possible loss of data [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_client.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\TCPClient.cpp(70,72): warning C4267: 'argument': conversion fro
m 'size_t' to 'int', possible loss of data [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_client.vcxpro
j]
MSVCRTD.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function "int __cdecl invoke_m
ain(void)" (?invoke_main@@YAHXZ) [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_client.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\Debug\tcp_client.exe : fatal error LNK1120: 1 unresolved exte
rnals [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_client.vcxproj]
  Building Custom Rule C:/Users/Harshitha/Desktop/PilotDataSynchronization/CMakeLists.txt
  TCPServer.cpp
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\TCPServer.cpp(42,30): warning C4244: '=': conversion from 'SOCK
ET' to 'int', possible loss of data [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_server.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\TCPServer.cpp(86,30): warning C4244: '=': conversion from 'SOCK
ET' to 'int', possible loss of data [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_server.vcxproj]
MSVCRTD.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function "int __cdecl invoke_m
ain(void)" (?invoke_main@@YAHXZ) [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_server.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\Debug\tcp_server.exe : fatal error LNK1120: 1 unresolved exte
rnals [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\tcp_server.vcxproj]
  Building Custom Rule C:/Users/Harshitha/Desktop/PilotDataSynchronization/CMakeLists.txt
  pilotdatasync-xp11.cpp
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(203,9): error C2872: 'byte': ambiguous symb
ol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(975,6): error C2872: 'byte': ambiguous symb
ol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(983,6): error C2872: 'byte': ambiguous symb
ol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypesbase.h(447,21): error C2872: 'byte': ambiguous
 symbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypesbase.h(472,21): error C2872: 'byte': ambiguous
 symbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypesbase.h(489,21): error C2872: 'byte': ambiguous
 symbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(85,21): error C2872: 'byte': ambiguous symb
ol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(94,21): error C2872: 'byte': ambiguous symb
ol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(100,21): error C2872: 'byte': ambiguous sym
bol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(106,21): error C2872: 'byte': ambiguous sym
bol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(112,21): error C2872: 'byte': ambiguous sym
bol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(118,21): error C2872: 'byte': ambiguous sym
bol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\wtypes.h(493,21): error C2872: 'byte': ambiguous sym
bol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidlbase.h(2275,74): error C2872: 'byte': ambiguous sy
mbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidlbase.h(2289,57): error C2872: 'byte': ambiguous sy
mbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidlbase.h(8492,74): error C2872: 'byte': ambiguous sy
mbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidlbase.h(8508,57): error C2872: 'byte': ambiguous sy
mbol [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(10444,64): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(11002,72): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(11034,72): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(11542,74): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(11557,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(12018,21): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(13832,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(13847,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15586,74): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15602,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15749,72): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15788,72): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15805,74): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(15822,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(16067,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\objidl.h(16086,57): error C2872: 'byte': ambiguous symbo
l [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\oaidl.h(571,21): error C2872: 'byte': ambiguous symbol [
C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
  (compiling source file '../src/pilotdatasync-xp11.cpp')
      C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared\rpcndr.h(202,23):
      could be 'unsigned char byte'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\cstddef(34,24):
      or       'std::byte'

C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(112,5): error C2065: 'TCPClient': undecl
ared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(112,15): error C2146: syntax error: miss
ing ';' before identifier 'client' [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(112,15): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(126,23): warning C4553: '==': result of
expression not used; did you intend '='? [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxpr
oj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(135,13): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(139,10): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(144,10): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(151,10): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(160,14): error C2065: 'client': undeclar
ed identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(231,5): error C2065: 'XPLMCreateFlightLo
op_t': undeclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(231,28): error C2146: syntax error: miss
ing ';' before identifier 'loop_params' [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxpro
j]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(231,28): error C2065: 'loop_params': und
eclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(232,30): error C2065: 'loop_params': und
eclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(233,18): error C2065: 'xplm_FlightLoop_P
hase_BeforeFlightModel': undeclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin
.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(232,9): error C7555: use of designated i
nitializers requires at least '/std:c++20' [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcx
proj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(231,40): error C3079: an initializer lis
t cannot be used as the right operand of this assignment operator [C:\Users\Harshitha\Desktop\PilotDataSynchronization\
build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(237,5): error C2065: 'XPLMFlightLoopID':
 undeclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(237,22): error C2146: syntax error: miss
ing ';' before identifier 'id' [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(237,22): error C2065: 'id': undeclared i
dentifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(237,49): error C2065: 'loop_params': und
eclared identifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(237,27): error C3861: 'XPLMCreateFlightL
oop': identifier not found [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(238,28): error C2065: 'id': undeclared i
dentifier [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(238,5): error C3861: 'XPLMScheduleFlight
Loop': identifier not found [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(170,5): warning C4996: 'strcpy': This fu
nction or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS.
 See online help for details. [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(171,5): warning C4996: 'strcpy': This fu
nction or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS.
 See online help for details. [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(172,5): warning C4996: 'strcpy': This fu
nction or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS.
 See online help for details. [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]
C:\Users\Harshitha\Desktop\PilotDataSynchronization\src\pilotdatasync-xp11.cpp(280,24): warning C4305: 'initializing':
truncation from 'double' to 'float' [C:\Users\Harshitha\Desktop\PilotDataSynchronization\build\xplane_plugin.vcxproj]

**Pull request closed without merging**